Source/items.cpp

```diff
@@ -1376,17 +1377,18 @@ int AllocateItem()
 
 static void GetSuperItemSpace(Point position, int8_t inum)
 {
-	if (GetItemSpace(position, inum))
+	Point positionToCheck = position;
+	if (GetItemSpace(positionToCheck, inum))
 		return;
 	for (int k = 2; k < 50; k++) {
 		for (int j = -k; j <= k; j++) {
-			int yy = position.y + j;
 			for (int i = -k; i <= k; i++) {
-				int xx = i + position.x;
-				if (!ItemSpaceOk(xx, yy))
+				Point offset = { i, j };
+				positionToCheck = position + offset;

```

Comment:
In other cases you doesn't seam to do temp variables, any reason it's different here?
```suggestion
				positionToCheck = position + Point { i, j };
```

Reply:
I felt it was more readable in this case to declare the offset.

Reply:
Ok.

On a side note it would be nice if we at some point generalize the code that searches 50 tiles since it's copied in a few places.

Reply:
Yeah I noticed it was duplicated while working on this... definitely needs some generalization. Perhaps a circular iterator for tiles where you specify how many tiles you want or the "radius" and it iterates over the tiles until those are reached.